### DISCUSSION TITLE
Discrete Drift Estimates

### DISCUSSION THREAD

**Discussion by SidneySaint** [cdriveraus/ctsem#31] [Q&A]
Hi Charles and community, 

I am wondering if there is a way to transform the confidence intervals for continuous drift estimates (from ctCI) into confidence intervals for discrete estimates (or another way of indexing the significance of the discrete drift coefficients)? (Please forgive me if there is an obvious answer to this question, my math background is quite weak.) 

Many thanks, 
Sidney

**Comment by cdriveraus** [comment]
Hi Sidney, you *may* want to ask yourself carefully if this is really what you want, but ctStanDiscretePars returns samples from the posterior and the desired quantiles can be computed, and the summary function has a $parmatrices section that includes dtDRIFT estimates (discrete time DRIFT), by default the time interval for this is 1 but can be adjusted using the timeinterval argument. Examples here:

```
library(data.table)
library(mltools)
library(ctsem)
chickdata <- one_hot(as.data.table(ChickWeight),cols = 'Diet')
chickdata$weight <- scale(chickdata$weight)
chickdata$Chick <- paste0(chickdata$Chick, "a") # Create Chick character id

m <- ctModel(
  LAMBDA=diag(1), #Factor loading matrix of latent processes on measurements, fixed to 1
  type = 'stanct', #Could specify 'standt' here for discrete time.
  tipredDefault = FALSE, #limit covariate effects on parameters to those explicitly specified
  manifestNames='weight', #Observed measurements of the latent processes
  latentNames='Lweight', #Names here simply make parameters and plots more interpretable
  TIpredNames = paste0('Diet_',2:4), #Covariates, in this case one category needs to be baseline...
  DRIFT='a11 | param', #normally self feedback (diagonal drift terms) are restricted to negative
  MANIFESTMEANS=0, #For identification CINT is normally zero with this freely estimated
  CINT='cint ||||Diet_2,Diet_3,Diet_4', #diet covariates specified in 5th 'slot' (four '|' separators)
  time='Time',
  id='Chick')

f <- ctStanFit(chickdata,m,nopriors=FALSE,cores=1)

summary(f,timeinterval = 2)
dp=ctStanDiscretePars(f,cores=1)
apply(dp,3,quantile,c(.025,.975))
```

**Reply by SidneySaint** [reply]
Thank you very much Charles! 
Sidney

**Reply by cdriveraus** [reply]
Oh, and you may want to increase nsamples for the ctStanDiscretePars function to 1000, for more precision.

**Reply by SidneySaint** [reply]
Thank you!
